src/CovidAppointmentTable.test.js

```diff
@@ -0,0 +1,73 @@
+import {
+  act,
+  render,
+  screen
+} from '@testing-library/react';
+
+import CovidAppointmentTable from './CovidAppointmentTable';
+
+import prod from '../test/fixtures/api/prod.json';
+import noData from '../test/fixtures/api/no-data.json';
+
+beforeAll(function() {
+  jest.spyOn(window, 'fetch');
+});
+
+describe('the CovidAppointmentTable component', function() {
+  describe('when api data is available', function() {
+    beforeEach(function() {
+      window.fetch.mockResolvedValueOnce({
+        json: async () => prod,
+        ok: true
+      });
+    });
+
+    test('it correctly displays available appointments', async function() {
+      await act(async function() {
+        render(
+          <CovidAppointmentTable />
+        );
+      });
+
+      expect(await screen.findByText(/Berkshire Community College Field House/)).toBeInTheDocument();
+    });
+  });
+
+  describe('when no api data is available', function() {
+    beforeEach(function() {
+      window.fetch.mockResolvedValueOnce({
+        json: async () => noData,
+        ok: true
+      });
+    });
+
+    test('it displays a loader component', async function() {
```

Comment:
I'm a little unclear here - are we testing that there's a "forever loader" essentially?